pkg/resolution/resource/name_test.go

```diff
@@ -113,3 +114,107 @@ func TestGenerateDeterministicName(t *testing.T) {
 		})
 	}
 }
+
+func TestGenerateDeterministicNameFromSpec(t *testing.T) {
+	type args struct {
+		prefix string
+		base   string
+		params []v1.Param
+		url    string
+	}
+	golden := args{
+		prefix: "prefix",
+		base:   "base",
+		params: []v1.Param{
+			{
+				Name: "string-param",
+				Value: v1.ParamValue{
+					Type:      v1.ParamTypeString,
+					StringVal: "value1",
+				},
+			},
+			{
+				Name: "array-param",
+				Value: v1.ParamValue{
+					Type:     v1.ParamTypeArray,
+					ArrayVal: []string{"value1", "value2"},
+				},
+			},
+			{
+				Name: "object-param",
+				Value: v1.ParamValue{
+					Type:      v1.ParamTypeObject,
+					ObjectVal: map[string]string{"key": "value"},
+				},
+			},
+		},
+		url: "https://foo/bar",
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "only contains prefix",
+			args: args{
+				prefix: golden.prefix,
+			},
+			want: "prefix-6c62272e07bb014262b821756295c58d",
+		},
+		{
+			name: "only contains base",
+			args: args{
+				base: golden.base,
+			},
+			want: "-6989337ae0757277b806e97e86444ef0",
+		},
+		{
+			name: "only contains url",
+			args: args{
+				url: golden.url,
+			},
+			want: "-dcfaf53735f4a84a3e319e17352940b4",
+		},
+		{
+			name: "only contains params",
+			args: args{
+				params: golden.params,
+			},
+			want: "-52921b17d3c2930a34419c618d6af0e9",
+		},
+		{
+			name: "params with different order should generate same hash",
+			args: args{
+				params: []v1.Param{
+					golden.params[2],
+					golden.params[1],
+					golden.params[0],
+				},
+			},
+			want: "-52921b17d3c2930a34419c618d6af0e9",
+		},
+		{
+			name: "contain all fields",
+			args: golden,
+			want: "prefix-ff25bd24688ab610bdc530a5ab3aabbd",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resolutionSpec := &v1beta1.ResolutionRequestSpec{
```

Comment:
Why does this span over 2 apiVersions?

Reply:
what do you mean? resolution request is still v1beta1.

Reply:
It's not the pipeline api. Its resolution api. Which is v1beta1.